src/main/scala/org/ergoplatform/nodeView/wallet/persistence/OffChainRegistry.scala

```diff
@@ -17,15 +18,15 @@ import scala.collection.mutable
   * @param onChainBalances  - on-chain balances snapshot (required to calculate off-chain indexes)
   */
 case class OffChainRegistry(height: Int,
-                            offChainBoxes: Seq[TrackedBox],
-                            onChainBalances: Seq[Balance]) {
+                            offChainBoxes: Array[TrackedBox],
+                            onChainBalances: Array[Balance]) {
```

Comment:
@jellymlg, using array fields in case classes breaks equality. I suggest to rollback this change.
It is always possible to use ArraySeq.unsafeWrapArray to put array into Seq.
As a general rule, Array should be avoided in fields and method singatures, but they still can be used as the actual implementation of Seq. This is both safe and also very fast.

Reply:
Reverted